test/python/test_ibmqjob_states.py

```diff
@@ -221,10 +222,10 @@ def test_never_complete_result_with_timeout(self):
         job = self.run_with_api(NonQueuedAPI())
 
         self.wait_for_initialization(job)
-        # We never make the API status to progress so it is stuck on RUNNING
-        with self.assertRaises(JobTimeoutError):
-            job.result(timeout=0.2)
-
+        result = job.result(timeout=0.2)
+        # TODO: This is inconsistent, the timeout prevents Qiskit from knowing
+        # the exact state of the job so it should raise.
+        self.assertRaises(QISKitError, result.get_data)
```

Comment:
It should be throwing `JobTimeoutError` here, right?

Reply:
It should, that way, we make it consistent again. But it's not because the error is not in the `Job` but in the `Result` object.